### PR TITLE
Fix for longer UserAgent suffix

### LINKED ISSUE #697: User-Agent string is too long and truncate customized user agent suffix
>We are continuously addressing and improving the SDK, if possible, make sure the problem persist in the [latest SDK version](https://www.nuget.org/packages/Microsoft.Azure.Cosmos).

**Describe the bug**
With 3.1.1, once we add customized User-Agent suffix to User-Agent field, from fidder I can see the whole User-Agent header got truncated. Like this:
User-Agent: Windows/10.0.17763 cosmos-netstandard-sdk/3.1.4 3.1.1-.NET Core 4.6.26614.01 X64 637014000798739244 cosmos-tab

We add "cosmos-table-sdk/2.0.0" to it but got truncated.

**Expected behavior**
User-Agent header should be optimized and leave enough space for customized user-agent suffix.

**Environment summary**
SDK Version: 3.1.1
OS Version (e.g. Windows, Linux, MacOSX) Windows

### PR DESCRIPTION
**Pull request by ealsur**
# Pull Request Template

## Description

This PR applies a refactor on the UserAgentContainer, creating a child class `Cosmos.UserAgentContainer` that has a different base user agent. The new base user agent uses `EnvironmentInformation` and generates a format of:

`{OS}/{Process Architecture} {SDK Version}/{Direct pkg Version}-{Framework} {ClientId} {UserSuffix}`

For example:

`Microsoft Windows 10.0.18362 /X64 3.1.1/3.1.4-.NET Core 4.6.27617.04 03445 mycustomsuffix`

The information is read from the RuntimeEnvironment NET Standard 2.0 https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.runtimeenvironment?view=netstandard-2.0

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

## Closing issues

Closes #697

### CODE REVIEW

**Comment by kirankumarkolli** [comment]
>Windows/10.0.17763 cosmos-netstandard-sdk/3.1.4 3.1.1-.NET Core 4.6.26614.01 X64 637014000798739244

Can it be abstracted to the scope we need. I.e. may-be 
> W/X64-WACosmos/3.1.... -.NET Core 4.6.26614.01

Thoughts?

**Comment by ealsur** [comment]
The problem is that the initial part comes as part of the UserAgentContainer defined in Direct package, which is used down the stack.

The alternative would be to redefine the class?

File: Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs [hunk 1]
```diff
@@ -85,8 +87,27 @@ public string ApplicationName
...
+
+                if (string.IsNullOrEmpty(value))
+                {
+                    return;
+                }
+
+                if (value.Length > UserAgentSuffixMaxLength)
+                {
+                    // Prioritize user suffix
+                    this.UserAgentContainer.Suffix = value;
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs]: We use this information a lot of the time to help troubleshoot issues. User might end up completely removing all the information we need. I agree we should shorten the user agent string even in the direct branch. Another possibility is introduce a new header like ClientVersion that we then record in kusto.

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs]: Refactored and removed this logic

**Comment by ealsur** [comment]
@kirankumarkolli I ended up refactoring this PR. The new introduced logic was too complex, so I ended up simplifying this by creating our own UserAgentContainer that inherit from UserAgentContainer from Direct but defines our own base user agent string, with the information we wanted (Package versions, etc)

**Comment by kirankumarkolli** [comment]
Can you please share new user-agent string?

**Comment by ealsur** [comment]
@kirankumarkolli it's in the PR description, I updated the description when I refactored

File: Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs [hunk 2]
```diff
@@ -62,10 +59,7 @@ public class CosmosClientOptions
         /// </summary>
         public CosmosClientOptions()
         {
-            this.UserAgentContainer = new UserAgentContainer();
-            EnvironmentInformation environmentInformation = new EnvironmentInformation();
-            this.currentEnvironmentInformation = environmentInformation.ToString();
-            this.UserAgentContainer.Suffix = this.currentEnvironmentInformation;
+            this.UserAgentContainer = new Cosmos.UserAgentContainer();
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs]: Does the central SDK team have any guidance on the format?

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs]: What I could gather is that they use a list of values: https://github.com/Azure/azure-sdk-for-net/blob/45cdad6bc3f20b36220d303a44f99895bdba28b7/sdk/mgmtcommon/ClientRuntime/ClientRuntime/ServiceClient.cs#L132 from this [ProductInfoHeaderValue](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.headers.productinfoheadervalue?view=netframework-4.8) base class. Each item contains a [ProductHeaderValue](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.headers.productheadervalue?view=netframework-4.8) and a string comment.
But these are HttpClient related classes.

There are no written recommendations though,

**Pull request merged**